### DISCUSSION TITLE
Boolean timeseries visualizers duplicate points

### DISCUSSION THREAD

**Discussion by jbscherrer** [bonsai-rx/bonsai#1543] [Q&A]
I use a lot of boolean timeseries visualizers in my workflows, and have noticed they exhibit an odd behavior. It seems that every boolean value generated by a node is represented in the visualizer by a pair of identical points one after the other, rather than by a single point (ie 101 is represented as 110011). Correspondingly, if I try to set the capacity of the visualizer, the number field autoupdates to twice the value that I enter. Is this behavior a bug, or is it somehow intentional? Thanks!

**Comment by glopesdev** [comment]
@jbscherrer It is "somewhat intentional" as you inferred. The reason is to make the plot look more like a square wave function you need 4 points to make the square, otherwise the plot would look like a triangular function.

Unfortunately due to the plotting framework we use (or the way we are using it at present) this detail bleeds into the plot config, but I agree it would be better to be more consistent. Happy to have an issue opened on this so will transfer this to an issue.

**Reply by jbscherrer** [reply]
Thanks Gonçalo! The intent makes sense, it just ends up being confusing if you aren't already aware of the behavior.

**Reply by bruno-f-cruz** [reply]
Not sure if this is relevant, but in case someone in the future wants to get rid of this behavior, a simple option is to cast the boolean to a numeric value, for instance a simple `ExpressionTransform` with `it ? 1 : 0` does the trick:

![image](https://github.com/bonsai-rx/bonsai/assets/7049351/a32766b2-1be7-4422-8f64-d0bad352dd18)

**Reply by jbscherrer** [reply]
That is helpful, thanks!
